app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1238,6 +1241,9 @@ private void setMapMarkerActions(Marker selectedMarker) {
     @Override
     public void recenterMap(fr.free.nrw.commons.location.LatLng curLatLng) {
         if (curLatLng == null) {
+            if (!locationManager.isNetworkProviderEnabled()) {
```

Comment:
Just like I expected this sort of brings us back to the [unintended behaviour that I described before](https://github.com/commons-app/apps-android-commons/pull/3438#discussion_r387212026). When I have the location mode set to "GPS only" and tap on the "Locate me" icon before the device is located I see the dialog. I thought we didn't want that.

In the following screenshot, note the location icon in the notification area indicating GPS is on.
![Screenshot_2020-03-17-23-24-23](https://user-images.githubusercontent.com/12448084/76888098-513dfc80-68a9-11ea-909d-40f54dc61b7c.png)

The following fixes this issue:
```suggestion
            if (!(locationManager.isNetworkProviderEnabled() || locationManager.isGPSProviderEnabled())) {
```

Reply:
> When I have the location mode set to "GPS only" and tap on the "Locate me" icon before the device is located I see the dialog. I thought we didn't want that.

No, we do want that. Before the device is located and it is on **GPS only**, it can't locate the device because **mobile** provider is not enabled hence the dialog.

Reply:
> No, we do want that.

I bet we don't. I'll explain the reason for why I'm stressing this in the end.

Anyways, I see some misunderstanding here. Let me try to clarify. [Previously](https://github.com/commons-app/apps-android-commons/pull/3438#issuecomment-599956481), when I mentioned Nearby works fine with "GPS only" I meant it works for me in *all cases*. So, it **works for me** even if it's the first time. Of course, it takes time depending on where you are. It's quick when you're outdoors. It takes time when you're inside a building as it's hard to find the satellites when you're indoors. As I said before, Nearby only wants the location to be enabled. Nothing more than that. If you observe otherwise, then it's an issue we should look into. Try going outdoors and observe if your device is located by Nearby when the location is in "GPS only". Let me know how this goes.

Coming to the reason about why we don't want to show the dialog whenever the network provider is disabled. Some privacy conscious people would prefer always to use "GPS only" mode for the Location as there's some controversy about privacy issues associated with what we call the network provider. It's a valid claim. If we show the dialog when the location is in "GPS only" mode and the network provider is disabled, then there's a chance those people might misinterpret this. It's best if we just avoid this.

Reply:
> Try going outdoors and observe if your device is located by Nearby when the location is in "GPS only". Let me know how this goes.

You are right! I tested Nearby with GPS only from my roof and it worked! I should have tested it outdoors :(